fetchmap/index.ts

```diff
@@ -0,0 +1,265 @@
+import './style.css';
+import maplibregl from 'maplibre-gl';
+import 'maplibre-gl/dist/maplibre-gl.css';
+import {Deck, Layer} from '@deck.gl/core';
+import {fetchMap, FetchMapResult, LayerDescriptor} from '@carto/api-client';
+import {BASEMAP} from '@deck.gl/carto';
+import {LayerFactory} from './utils';
+import {createLegend} from './legend';
+import './legend.css';
+import {buildTooltip} from './tooltip';
+
+const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+
+// Define configurations for multiple maps
+const MAP_CONFIGS = [
+  {id: '26bce5fe-29d4-48dc-914f-14355971f143', name: 'US Weather'},
+  {id: '6ac6f3bb-9c90-4e7e-9ecd-3c7343704c24', name: 'US Wind Map'}
+];
+
+// Define the structure for aggregated map data
+interface AggregatedMapData {
+  title: string;
+  layers: LayerDescriptor[];
+  popupSettings: FetchMapResult['popupSettings'] | null;
+  initialViewState?: Deck['props']['initialViewState'];
+}
+
+let currentMapData: AggregatedMapData | null = null; // Stores aggregated data from selected maps
+
+// Base options for fetchMap, cartoMapId will be added per call
+const baseFetchMapOptions = {
+  apiBaseUrl
+  // accessToken: import.meta.env.VITE_CARTO_ACCESS_TOKEN // if you want to use a private (non-public) map
+};
+
+// HTML Elements
+const mapNameEl = document.querySelector<HTMLDListElement>('#mapName');
+const layerCountEl = document.querySelector<HTMLDListElement>('#layerCount');
+const mapSelectorContainer = document.getElementById('map-selector-container');
+const loadMapsButton = document.getElementById('load-maps-button');
```

Comment:
For cases like this where we 'know' the element is on the page, I think it's a worthwhile simplification to use a non-null assertion...

```diff
-const mapNameEl = document.querySelector<HTMLDListElement>('#mapName');
+const mapNameEl = document.querySelector<HTMLDListElement>('#mapName')!;
```

... and then later use of the element is cleaner:

```diff
-if (mapNameEl) mapNameEl.innerHTML = 'No Map Selected';
+mapNameEl.innerHTML = 'No Map Selected';
```

Might be more readable to use just one of querySelector or getElementById here too.